File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    if(s.length % 2 !== 0) return false;
    let stack = [];

    for(let char of s){
        if(char === '(') stack.push(')');
        else if(char === '[') stack.push(']');
        else if(char === '{') stack.push('}');
        else{
            if(s.length === 0 || stack.pop() !== char) return false;
        }
    }

    return stack.length === 0;
};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        if len(s) % 2 != 0:
            return False
        
        stack = []
        
        for char in s:
            if char == '(':
                stack.append(')')
            elif char == '[':
                stack.append(']')
            elif char == '{':
                stack.append('}')
            else:
                if not stack or stack.pop() != char:
                    return False
                    
        return len(stack) == 0